packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -95,9 +96,10 @@ export default function assets({
 					}
 
 					// Rollup will copy the file to the output directory, this refer to this final path, not to the original path
-					const finalOriginalImagePath = (
-						isESMImportedImage(options.src) ? options.src.src : options.src
-					).replace(settings.config.build.assetsPrefix || '', '');
+					const ESMImportedImageSrc = isESMImportedImage(options.src) ? options.src.src : options.src;
+					const fileExtension = extname(ESMImportedImageSrc);
+					const pf = getAssetsPrefix(fileExtension, settings.config.build.assetsPrefix);
```

Comment:
I think would not be prone to bugs.

`defaultAssetsPrefix` is default assets prefix cdn value set. You can set the default values of other files by setting the value of `defaultAssetsPrefix`. If `defaultAssetsPrefix` is not set, when the file extension is not found in the `assetsPrefix` object, it will be set to the empty string `''` by default.

Reply:
I think it would. Let's say we decide to rename the output from `.js` to `.mjs` today, that would mean the existing configuration won't work anymore. If someone wants to be robust, they have to configure `{ js: '...', mjs: '...', cjs: '...' }` which I don't think is a nice DX. We can help here by grouping them up as a single asset type.